### PR TITLE
Bug 2057101: remove klog format and update messages for docker config deprecation

### PR DESCRIPTION
**Pull request by atiratree**

### CODE REVIEW

File: pkg/helpers/image/helpers.go [hunk 1]
```diff
@@ -148,7 +148,7 @@ func GetRegistryAuthConfigPreference() RegistryAuthConfigPreference {
 		result = RegistryAuthConfigPreference(authPreference)
 	} else {
 		// TODO: remove once deprecated in 4.12
-		klog.Warningln("Defaulting of registry auth file to \"${HOME}/.docker/config.json\" is deprecated. The default will be switched to podman config locations in the future version.")
+		fmt.Fprintf(os.Stderr, "Warning: the default reading order of registry auth file will be changed from \"${HOME}/.docker/config.json\" to podman registry config locations in the future version. \"${HOME}/.docker/config.json\" is deprecated, but can still be used for storing credentials as a fallback.\n")
```
Comment (soltysh) [inline pkg/helpers/image/helpers.go:151]: Can you add more info (maybe a link) to what the podman registry order means?

**Review by soltysh — CHANGES_REQUESTED** [review]

File: pkg/helpers/image/helpers.go (hunk 1 above)
Comment (atiratree) [reply pkg/helpers/image/helpers.go]: @soltysh Yup, added a link

**Review by soltysh — APPROVED** [review]
/lgtm
/approve
/retest

**Comment by cblecker** [comment]
This is happening on the current 4.10 CLI, so requesting a cherry-pick back

/cherry-pick release-4.10

**Pull request merged**